src/ImageManipulation.php

```diff
@@ -1057,4 +1062,98 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    private function copyImageBackend()
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $file->setImageBackend($this->getImageBackend());
+        return $file->setOriginal($this);
+    }
+
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->setMutableAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    public function setMutableAttributes($attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    protected function getDefaultAttributes(): array
+    {
+        if (!$this->getIsImage()) {
+            return [];
+        }
+
+        $attributes = [
+            'width' => $this->getWidth(),
+            'height' => $this->getHeight()
+        ];
+
+        if ($this->getIsLazyLoaded()) {
+            $attributes['loading'] = 'lazy';
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * Return whether image will be lazy loaded
+     *
+     * @return bool
+     */
+    public function getIsLazyLoaded() : bool
+    {
+        if (Image::getLazyLoadingEnabled() && $this->getIsImage()) {
+            return empty($this->attributes['loading']) || $this->attributes['loading'] === 'lazy';
+        }
+        return false;
+    }
+
+    public function LazyLoad($lazyLoad): AssetContainer
+    {
+        if (!Image::getLazyLoadingEnabled() || !$this->getIsImage()) {
+            return $this;
+        }
+
+        if (is_string($lazyLoad)) {
+            $lazyLoad = strtolower($lazyLoad);
+        }
+
+        switch ($lazyLoad) {
+            case false:
+            case 0:
+            case '0':
+            case 'false':
+            case 'eager':
+                return $this->setAttribute('loading', 'eager');
```

Comment:
A side affect of this is that we will always add `loading="eager"` when we disable lazy loading on a specific image.

If we thing this is a problem, I can probably find a way to not output the attribute at all.

Reply:
Thinking so far has been to omit the loading attribute entirely when eager loading.  Seems like all the online articles talking about recommend that it's omitted

If we changes this to a `removeAttribute('loading)` call would everything (immutable/chainable) work as it should?

Reply:
I'll get it removed. Just need to tweak the `getAttributes` method a bit.

Reply:
I've added a bit of logic so we don't output `loading="eager"`